Add docs about using JavaScript libraries that render UI (#20091)
dotnet/AspNetCore.Docs: aspnetcore/blazor/call-javascript-from-dotnet.md

}
```

## Use of JavaScript libraries that render UI (DOM elements)

Sometimes you may wish to use JavaScript libraries that produce visible user interface elements within the browser DOM. At first glance, this might seem difficult because Blazor's diffing system relies on having control over the tree of DOM elements and runs into errors if some external code mutates the DOM tree and invalidates its mechanism for applying diffs. This isn't a Blazor-specific limitation. The same challenge occurs with any diff-based UI framework.

> Fortunately, it's straightforward to embed externally-generated UI within a Blazor component UI reliably. The recommended technique is to have the component's code (`.razor` file) produce an empty element. As far as Blazor's diffing system is concerned, the element is always empty, so the component isn't required to recurse into the element and  leave its contents alone. This makes it safe to populate the element with arbitrary externally-managed content.

Author: One part of this edit looks erroneous:

```
so the component isn't required to recurse into the element and  leave its contents alone
```

This has the opposite meaning than intended. We do want it to leave the contents alone, but this says "it isn't required to leave its contents alone".

```suggestion
Fortunately, it's straightforward to embed externally-generated UI within a Blazor component UI reliably. The recommended technique is to have the component's code (`.razor` file) produce an empty element. As far as Blazor's diffing system is concerned, the element is always empty, so the renderer does not recurse into the element and instead leaves its contents alone. This makes it safe to populate the element with arbitrary externally-managed content.
```

The corresponding JavaScript module, which should be placed at `wwwroot/mapComponent.js`, is as follows:

```javascript
import 'https://api.mapbox.com/mapbox-gl-js/v1.12.0/mapbox-gl.js';

> mapboxgl.accessToken = '{ACCESS TOKEN}';

Author: I don't think you should remove the comment (`// TO MAKE THE MAP APPEAR...`). This is taken directly from https://docs.mapbox.com/mapbox-gl-js/api/.

The information about where to get an API token (https://account.mapbox.com) is essential for anyone who wants to make this code run correctly.

Reviewer: It's below ... we're trying to get text out of code examples. See if the paragraph that follows covers it.

How about both? I'll add it back. I just want to make sure that we get a localized version of that guidance in the text.

I'll do it on the next commit.

Author: Ah OK, thanks. Both sounds good!

Reviewer: Done! :+1:

export function addMapToElement(element) {
  return new mapboxgl.Map({
    container: element,
    style: 'mapbox://styles/mapbox/streets-v11',
    center: [-74.5, 40],
    zoom: 9
  });
}

export function setMapCenter(map, latitude, longitude) {
  map.setCenter([longitude, latitude]);
}
```

> The placeholder `{ACCESS TOKEN}` in the preceding code is a valid access token from https://account.mapbox.com.

Author: Not sure about this phrasing! It sounds to me like "You can use the string {ACCESS TOKEN} as an access token", but you can't. Could we emphasise something like "You need to replace the string {ACCESS TOKEN} with a valid access token that you can get from ..."
